Apps/WebClient/src/Server/Services/WalletStatusService.cs

```diff
@@ -0,0 +1,116 @@
+    using HealthGateway.Common.ErrorHandling;
+    using HealthGateway.Common.Models;
+    using HealthGateway.Database.Constants;
+    using HealthGateway.Database.Delegates;
+    using HealthGateway.Database.Models;
+    using HealthGateway.Database.Wrapper;
+    using Microsoft.Extensions.Logging;
+
+    /// <inheritdoc />
+    public class WalletStatusService : IWalletStatusService
+    {
+        private readonly ILogger logger;
+        private readonly IWalletDelegate walletDelegate;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="WalletStatusService"/> class.
+        /// </summary>
+        /// <param name="logger">Injected Logger Provider.</param>
+        /// <param name="walletDelegate">Injected Wallet delegate.</param>
+        public WalletStatusService(ILogger<WalletStatusService> logger, IWalletDelegate walletDelegate)
+        {
+            this.logger = logger;
+            this.walletDelegate = walletDelegate;
+        }
+
+        /// <inheritdoc />
+        public RequestResult<WalletConnection> UpdateWalletConnection(Guid connectionId)
+        {
+            RequestResult<WalletConnection> result = new ()
+            {
+                ResultStatus = Common.Constants.ResultType.Error,
+            };
+
+            DBResult<WalletConnection> dbResult = this.walletDelegate.GetConnection(connectionId);
+            if (dbResult.Status == DBStatusCode.Read)
+            {
+                WalletConnection connection = dbResult.Payload;
+                result.ResourcePayload = connection;
+                connection.ConnectedDateTime = DateTime.UtcNow;
+                connection.Status = WalletConnectionStatus.Connected;
+                DBResult<WalletConnection> updateResult = this.walletDelegate.UpdateConnection(connection, true);
+                if (updateResult.Status == DBStatusCode.Updated)
+                {
+                    result.ResultStatus = Common.Constants.ResultType.Success;
+                    result.TotalResultCount = 1;
+                }
+                else
+                {
+                    this.logger.LogWarning($"Unable to update wallet connection with id: {connectionId}");
+                    result.ResultError = new RequestResultError() { ResultMessage = "Error Updating Wallet Connection", ErrorCode = ErrorTranslator.ServiceError(ErrorType.CommunicationExternal, ServiceType.WalletIssuer) };
+                }
+            }
+            else
+            {
+                this.logger.LogWarning($"Unable to find wallet connection with id: {connectionId}");
+                result.ResultError = new RequestResultError() { ResultMessage = "Unable to find Wallet Connection", ErrorCode = ErrorTranslator.ServiceError(ErrorType.CommunicationExternal, ServiceType.WalletIssuer) };
+            }
+
+            return result;
+        }
+
+        /// <inheritdoc />
+        public RequestResult<WalletCredential> UpdateWalletCredential(Guid exchangeId)
```

Comment:
Likewise, if this only handles setting the credential status to "Added", maybe we should use a less-generic name for the method.

Reply:
I made them precise and and to take a state to reflect what they are doing.